tests/parametric/test_otel_span_methods.py

```diff
@@ -330,21 +588,797 @@ def test_otel_set_span_status_ok(self, test_agent, test_library):
+            - messaging.operation is set to something
+
+            (https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/messaging/)
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_SERVER) as span:
+                span.set_attributes({"messaging.system": "kafka"})
+                span.set_attributes({"messaging.operation": "receive"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, otel_span(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "kafka.receive"
+        assert root_span["resource"] == "otel_span_name"
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_operation_name_aws_client_001(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to `"aws." + `rpc.service`.lower() + ".request" :
+            - Span kind is set to Client
+            - rpc.system is set to aws-api
+            - rpc.service is set to something (e.g., "S3" in this example)
+
+            (https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/instrumentation/aws-sdk/)
+        """
+        with test_library:
+            with test_library.otel_start_span("otel_span_name", span_kind=SK_CLIENT) as span:
+                span.set_attributes({"rpc.system": "aws-api"})
+                span.set_attributes({"rpc.service": "S3"})
+                span.end_span()
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, otel_span(name="otel_span_name"))
+        assert len(trace) == 1
+
+        root_span = get_span(test_agent)
+
+        assert root_span["name"] == "aws.s3.request"
+        assert root_span["resource"] == "otel_span_name"
+
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_operation_name_aws_client_002(self, test_agent, test_library):
+        """
+            Tests that the operation name will be set to `"aws.request" :
```

Comment:
Looking at the sample code again, I think this should be `aws.client.request` to align with the others in that case. LMK if you disagree. I've gone ahead and updated the doc to reflect this.

Reply:
Thanks, updated in a109769a3d4df75228281f351d03035e2900fad9